hack/run-integration-tests: uses installed krew

hack/run-integration-tests.sh does
```

KREW_BINARY=$(readlink -f "${1:-$KREW_BINARY_DEFAULT}")  # needed for `kubectl krew` in tests
if [[ ! -x "${KREW_BINARY}" ]]; then
  echo "Did not find $KREW_BINARY. You need to build krew before running the integration tests"
  exit 1
fi
export KREW_BINARY
```

however it still calls `kubectl krew` under the covers, which for some reason isn't picking up `$KREW_BINARY`.

For example, when I have krew installed, an run this script, I see tests fail:


```
--- FAIL: TestKrewUninstall (1.34s)
    krew_test.go:70: krew [uninstall konfig]: exit status 255, F0625 16:52:31.788014   31000 root.go:52] unknown command "uninstall" for "krew"

        Did you mean this?
        	install
```

--- a/hack/ensure-kubectl-installed.sh
+++ b/hack/ensure-kubectl-installed.sh
@@ -29,6 +29,7 @@ install_kubectl_if_needed() {
     goarch="$(go env GOARCH)"
     kubectl_url="https://storage.googleapis.com/kubernetes-release/release/${kubectl_version}/bin/${goos}/${goarch}/kubectl"
 
+    echo >&2 "kubectl not detected in environment, downloading ${kubectl_url}"
     mkdir -p "${bin_dir}"
     curl --fail --show-error --silent --location --output "$kubectl_path" "${kubectl_url}"
     chmod +x "$kubectl_path"

--- a/hack/run-integration-tests.sh
+++ b/hack/run-integration-tests.sh
@@ -38,5 +38,4 @@ if [[ ! -x "${KREW_BINARY}" ]]; then
 fi
 export KREW_BINARY
 
-"${SCRIPTDIR}/ensure-kubectl-installed.sh"
 go test -test.v sigs.k8s.io/krew/test

--- a/hack/run-integration-tests.sh
+++ b/hack/run-integration-tests.sh
@@ -18,9 +18,9 @@ set -euo pipefail
 
 SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 BINDIR="${SCRIPTDIR}/../out/bin"
-GOOS="$(go env GOOS)"
-GOARCH="$(go env GOARCH)"
-KREW_BINARY_DEFAULT="$BINDIR/krew-${GOOS}_${GOARCH}"
+goos="$(go env GOOS)"
+goarch="$(go env GOARCH)"
+KREW_BINARY_DEFAULT="${BINDIR}/krew-${goos}_${goarch}"
 
 if [[ "$#" -gt 0 && ( "$1" = '-h' || "$1" = '--help' ) ]]; then
   cat <<EOF
@@ -33,9 +33,10 @@ fi
 
 KREW_BINARY=$(readlink -f "${1:-$KREW_BINARY_DEFAULT}")  # needed for `kubectl krew` in tests
 if [[ ! -x "${KREW_BINARY}" ]]; then
-  echo "Did not find $KREW_BINARY. You need to build krew before running the integration tests"
+  echo "Did not find $KREW_BINARY. You need to build krew FOR ${goos}/${goarch} before running the integration tests."
   exit 1
 fi
 export KREW_BINARY
 
-go test -v ./...
+"${SCRIPTDIR}/ensure-kubectl-installed.sh"
+go test -test.v sigs.k8s.io/krew/test

--- a/test/krew/krew.go
+++ b/test/krew/krew.go
@@ -19,6 +19,7 @@ import (
 	"fmt"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"testing"
 	"time"
 
@@ -32,51 +33,82 @@ const krewBinaryEnv = "KREW_BINARY"
 
 // ITest is used to set up `krew` integration tests.
 type ITest struct {
-	t       *testing.T
-	plugin  string
-	args    []string
-	env     []string
-	tempDir *testutil.TempDir
+	t             *testing.T
+	plugin        string
+	pluginsBinDir string
+	args          []string
+	env           []string
+	tempDir       *testutil.TempDir
 }
 
 // NewTest creates a fluent krew ITest.
 func NewTest(t *testing.T) (*ITest, func()) {
 	tempDir, cleanup := testutil.NewTempDir(t)
-	pathEnv := setupPathEnv(t, tempDir)
+	binDir := setupKrewBin(t, tempDir)
+
 	return &ITest{
-		t:       t,
-		env:     []string{pathEnv, fmt.Sprintf("KREW_ROOT=%s", tempDir.Root())},
+		t:             t,
+		pluginsBinDir: binDir,
+		env: []string{
+			fmt.Sprintf("KREW_ROOT=%s", tempDir.Root()),
+			fmt.Sprintf("PATH=%s", augmentPATH(t, binDir)),
+		},
 		tempDir: tempDir,
 	}, cleanup
 }
 
-func setupPathEnv(t *testing.T, tempDir *testutil.TempDir) string {
-	krewBinPath := tempDir.Path("bin")
-	if err := os.MkdirAll(krewBinPath, os.ModePerm); err != nil {
+// setupKrewBin symlinks the $KREW_BINARY to $tempDir/bin and returns the path
+// to this directory.
+func setupKrewBin(t *testing.T, tempDir *testutil.TempDir) string {
+	krewBinary, found := os.LookupEnv(krewBinaryEnv)
+	if !found {
+		t.Fatalf("%s environment variable pointing to krew binary not set", krewBinaryEnv)
+	}
+	binPath := tempDir.Path("bin")
+	if err := os.MkdirAll(binPath, 0755); err != nil {
 		t.Fatal(err)
 	}
-
-	if krewBinary, found := os.LookupEnv(krewBinaryEnv); found {
-		if err := os.Symlink(krewBinary, tempDir.Path("bin/kubectl-krew")); err != nil {
-			t.Fatalf("Cannot link to krew: %s", err)
-		}
-	} else {
-		t.Logf("Environment variable %q was not found, using krew installation from host", krewBinaryEnv)
+	if err := os.Symlink(krewBinary, filepath.Join(binPath, "kubectl-krew")); err != nil {
+		t.Fatalf("cannot link krew binary: %s", err)
 	}
-
-	path, found := os.LookupEnv("PATH")
-	if !found {
-		t.Fatalf("PATH variable is not set up")
-	}
-
-	return fmt.Sprintf("PATH=%s:%s", krewBinPath, path)
+	return binPath
 }
 
-// Call configures the runner to call plugin with arguments args.
-func (it *ITest) Call(plugin string, args ...string) *ITest {
-	it.plugin = plugin
-	it.args = args
-	return it
+// augmentPATH apprends the value to the current $PATH and returns the new
+// value.
+func augmentPATH(t *testing.T, v string) string {
+	curPath, found := os.LookupEnv("PATH")
+	if !found {
+		t.Fatalf("$PATH variable is not set up, required to run tests")
+	}
+
+	return v + string(os.PathListSeparator) + curPath
+}
+
+func (it *ITest) lookExecutable(file string) error {
+	orig := os.Getenv("PATH")
+	defer func() { os.Setenv("PATH", orig) }()
+
+	binPath := filepath.Join(it.Root(), "bin")
+	os.Setenv("PATH", binPath+string(os.PathListSeparator)+binPath)
+
+	_, err := exec.LookPath(file)
+	return err
+}
+
+// AssertExecutableInPATH asserts that the executable file is in bin path.
+func (it *ITest) AssertExecutableInPATH(file string) {
+	if err := it.lookExecutable(file); err != nil {
+		it.t.Fatalf("executable %s not in PATH: %+v", file, err)
+	}
+}
+
+// AssertExecutableNotInPATH asserts that the executable file is not in bin
+// path.
+func (it *ITest) AssertExecutableNotInPATH(file string) {
+	if err := it.lookExecutable(file); err == nil {
+		it.t.Fatalf("executable %s still exists in PATH", file)
+	}
 }
 
 // Krew configures the runner to call krew with arguments args.
@@ -155,8 +187,7 @@ func (it *ITest) cmd(ctx context.Context) *exec.Cmd {
 	args = append(args, it.args...)
 
 	cmd := exec.CommandContext(ctx, "kubectl", args...)
-	cmd.Env = append(os.Environ(), it.env...)
-
+	cmd.Env = it.env // clear env, do not inherit from system
 	return cmd
 }
 

--- a/test/krew/krew.go
+++ b/test/krew/krew.go
@@ -85,12 +85,12 @@ func augmentPATH(t *testing.T, v string) string {
 	return v + string(os.PathListSeparator) + curPath
 }
 
-func (it *ITest) lookExecutable(file string) error {
+func (it *ITest) lookupExecutable(file string) error {
 	orig := os.Getenv("PATH")
 	defer func() { os.Setenv("PATH", orig) }()
 
 	binPath := filepath.Join(it.Root(), "bin")
-	os.Setenv("PATH", binPath+string(os.PathListSeparator)+binPath)
+	os.Setenv("PATH", binPath)
 
 	_, err := exec.LookPath(file)
 	return err
@@ -98,7 +98,8 @@ func (it *ITest) lookExecutable(file string) error {
 
 // AssertExecutableInPATH asserts that the executable file is in bin path.
 func (it *ITest) AssertExecutableInPATH(file string) {
-	if err := it.lookExecutable(file); err != nil {
+	it.t.Helper()
+	if err := it.lookupExecutable(file); err != nil {
 		it.t.Fatalf("executable %s not in PATH: %+v", file, err)
 	}
 }
@@ -106,7 +107,8 @@ func (it *ITest) AssertExecutableInPATH(file string) {
 // AssertExecutableNotInPATH asserts that the executable file is not in bin
 // path.
 func (it *ITest) AssertExecutableNotInPATH(file string) {
-	if err := it.lookExecutable(file); err == nil {
+	it.t.Helper()
+	if err := it.lookupExecutable(file); err == nil {
 		it.t.Fatalf("executable %s still exists in PATH", file)
 	}
 }

--- a/test/krew_test.go
+++ b/test/krew_test.go
@@ -57,7 +57,7 @@ func TestKrewInstall(t *testing.T) {
 	defer cleanup()
 
 	test.WithIndex().Krew("install", validPlugin).RunOrFailOutput()
-	test.Call(validPlugin, "--help").RunOrFail()
+	test.AssertExecutableInPATH("kubectl-" + validPlugin)
 }
 
 func TestKrewUninstall(t *testing.T) {
@@ -68,9 +68,7 @@ func TestKrewUninstall(t *testing.T) {
 
 	test.WithIndex().Krew("install", validPlugin).RunOrFailOutput()
 	test.Krew("uninstall", validPlugin).RunOrFailOutput()
-	if err := test.Call(validPlugin, "--help").Run(); err == nil {
-		t.Errorf("Expected the plugin to be uninstalled")
-	}
+	test.AssertExecutableNotInPATH("kubectl-" + validPlugin)
 }
 
 func TestKrewSearchAll(t *testing.T) {

